java/query/GraqlDelete.java

```diff
@@ -19,108 +19,77 @@
- * properties are deleted.
+ * are provided to it. Only those properties will be deleted.
  */
-public class GraqlDelete extends GraqlQuery implements Filterable {
+public class GraqlDelete extends GraqlQuery {
 
     private final MatchClause match;
-    private final LinkedHashSet<Variable> vars;
-    private final Filterable.Sorting sorting;
-    private final long offset;
-    private final long limit;
+    private final List<Statement> statements;
 
-    public GraqlDelete(MatchClause match) {
-        this(match, new LinkedHashSet<>());
-    }
-
-    public GraqlDelete(MatchClause match, LinkedHashSet<Variable> vars) {
-        this(match, vars, null, -1, -1);
-    }
-
-    public GraqlDelete(MatchClause match, LinkedHashSet<Variable> vars, Filterable.Sorting sorting, long offset, long limit) {
+    public GraqlDelete(MatchClause match, List<Statement> statements) {
         if (match == null) {
             throw new NullPointerException("Null match");
         }
         this.match = match;
 
-        if (vars == null) {
-            throw new NullPointerException("Null vars");
+        if (statements == null) {
+            throw new NullPointerException("Null delete");
         }
-        for (Variable var : vars) {
-            if (!match.getSelectedNames().contains(var)) {
-                throw GraqlException.variableOutOfScope(var.toString());
-            }
-        }
-        this.vars = vars;
 
-        if (sorting != null && !vars().contains(sorting.var())) {
-            throw GraqlException.variableOutOfScope(sorting.var().toString());
+        if (statements.isEmpty()) {
+            throw GraqlException.noPatterns();
         }
-        this.sorting = sorting;
-        this.offset = offset;
-        this.limit = limit;
+
+        statements.forEach(statement ->{
+            for (Variable var : statement.variables()) {
+                if (!match.getSelectedNames().contains(var)) {
+                    throw GraqlException.deleteVariableUnbound(var.toString());
+                }
+            }
+        });
+        this.statements = statements;
     }
 
     @CheckReturnValue
     public MatchClause match() {
         return match;
     }
 
-    @CheckReturnValue // TODO: Return LinkedHashSet
-    public Set<Variable> vars() {
-        if (vars.isEmpty()) return match.getPatterns().variables();
-        return vars;
-    }
-
-    @Override
-    public Optional<Sorting> sort() {
-        return Optional.ofNullable(sorting);
+    public List<Statement> statements() {
+        return statements;
     }
 
-    @Override
-    public Optional<Long> offset(){
-        return this.offset < 0 ? Optional.empty() : Optional.of(this.offset);
-    }
-
-    @Override
-    public Optional<Long> limit() {
-        return limit < 0 ? Optional.empty() : Optional.of(limit);
+    public Set<Variable> vars() {
+        return statements.stream().flatMap(statement -> statement.variables().stream()).collect(Collectors.toSet());
```

Comment:
I should have caught it during the review, but we shouldn't have kept `vars()` in `GraqlDelete`, @flyingsilverfin . Given that it now takes in "instance statements", it should only be holding onto those, not caring about `vars()` and the equality check should be based on `statements` too.